data/campaigns/Under_the_Burning_Suns/scenarios/09_Blood_is_Thicker_Than_Water.cfg

```diff
@@ -1842,31 +1838,27 @@
             message= _ "Forgive me, Kaleh. I do not know what to believe. I... I have to ponder this."
         [/message]
 
-        {MESSAGE_DEPEND_ON_ALLY
-        (
-            [message]
-                speaker=Grog
-                message= _ "Great Leader told Grog to serve you, and so Grog will still follow your command. But other elves must be out on a separate island. How will we cross deep water?"
-            [/message]
-        )
-        (
-            [message]
-                speaker=Nog
-                message= _ "Great Leader told Nog to serve you, and so Nog will still follow your command. But other elves must be out on a separate island. How will we cross deep water?"
-            [/message]
-        )
-        (
-            [message]
-                speaker=Rogrimir
-                message= _ "The chieftain told me to serve you, and it will take more than this to shatter my confidence in you, lad. But those elves must be out on a different island, how will we cross the deep water?"
-            [/message]
-        )
-        (
-            [message]
-                speaker=Jarl
-                message= _ "The chieftain told me to serve you, and it will take more than this to shatter my confidence in you, lad. But those elves must be out on a different island, how will we cross the deep water?"
-            [/message]
-        )}
+        [if]
+            [variable]
+                name=ally_race
+                equals=dwarf
+            [/variable]
+
+            [then]
+                [message]
+                    speaker=$ally_id
+                    message= _ "The chieftain told me to serve you, and it will take more than this to shatter my confidence in you, lad. But those elves must be out on a different island, how will we cross the deep water?"
+                [/message]
+            [/then]
+
+            [else]
+                [message]
+                    speaker=$ally_id
+                    message= _ "Great Leader told $ally_unit.name to serve you, and so $ally_unit.name will still follow your command. But other elves must be out on a separate island. How will we cross deep water?"
```

Comment:
With the names taken out, lines like this should have a `# po: speaker is a troll` hint.

Reply:
Done.